tests/phpunit/tests/block-template-utils.php

```diff
@@ -321,6 +339,62 @@ static function( $template ) {
 		*/
 	}
 
+	/**
+	 * @dataProvider data_get_block_template_should_respect_posttypes_property
+	 * @ticket 55881
+	 * @covers ::get_block_templates
+	 *
+	 * @param string $post_type Post type for query.
+	 * @param array  $expected  Expected template IDs.
+	 */
+	public function test_get_block_template_should_respect_posttypes_property( $post_type, $expected ) {
```

Comment:
To isolate the tests for the fix, I've:

* created each dataset for switching the post type
* encapsulated the tests for validating that it respects the post type property

By isolating the tests, it makes it easier to identify the purpose of the test. It will also will make it easier when the test class is split into separate files one per function (this is out-of-scope for this PR).

The test class itself needs to be split into separate files, one for each function. But that's out-of-scope for this PR.